tests/integration/test_filters.py

```diff
@@ -58,6 +58,55 @@ def test_filters_are_applied(
         events_consumer.get_event()
 
 
+@pytest.mark.parametrize(
+    "is_processing_relay", (False, True), ids=["external relay", "processing relay"]
+)
+@pytest.mark.parametrize(
+    "event, must_filter",
+    (
+        ({"logentry": {"message": "filter:"}}, False),
+        ({"logentry": {"message": "filter:yes"}}, True),
+        ({"logentry": {"message": "filter:%s", "params": ["no"]}}, False),
+        ({"logentry": {"message": "filter:%s", "params": ["yes"]}}, True),
+    ),
+    ids=[
+        "plain non-matching message",
+        "plain matching message",
+        "formatting non-matching message",
+        "formatting matching message",
+    ],
+)
+def test_error_message_filters_are_applied(
+    mini_sentry,
+    events_consumer,
+    relay,
+    relay_with_processing,
+    is_processing_relay,
+    event,
+    must_filter,
+):
+    events_consumer = events_consumer()
+    processing = relay_with_processing()
+    if is_processing_relay:
+        relay = processing
+    else:
+        relay = relay(processing)
+
+    project_id = 42
+    project_config = mini_sentry.add_full_project_config(project_id)
+    project_config["config"]["filterSettings"]["errorMessages"] = {
+        "patterns": ["*yes*"]
+    }
+
+    relay.send_event(project_id, event)
+
+    if must_filter:
+        events_consumer.assert_empty()
+    else:
+        events_consumer.get_event()
+        events_consumer.assert_empty()
```

Comment:
I ran this parametrized test on master, and all 8 instances of it passed, so there might still be something wrong with the hypothesis.

Reply:
I added a 1s wait time after sending the event, in bf1d39c0f783d1b1cc7f6017e53a1cdce1abe66e. When I was testing it manually, I saw a successful `assert_empty` followed by a `get_event` retrieving the event. I'm not sure what is causing that, but adding that 1s addressed this issue (passing here, failing on master).

Why 1s? 0.5s isn't enough, and 2s seems too much for now (we can reconsider this if it causes flakiness).